backend/uclapi/workspaces/models.py

```diff
@@ -1,7 +1,88 @@
 """
-Models for Workspaces app. All commented out to keep
-Hound happy until this file is used in the future.
+Models for Workspaces app
 """
-# from django.db import models
+from django.db import models
+from datetime import datetime
 
-# Create your models here.
+
+class Sensors(models.Model):
+    sensor_id = models.IntegerField()
+    survey_id = models.IntegerField()
+    hardware_id = models.IntegerField()
+    survey_device_id = models.IntegerField()
+
+    def __str__(self):
+        return f"Sensors object, {self.sensor_id} @ {self.survey_id}"
+
+    class Meta:
+        _DATABASE = 'default'
+        unique_together = ('sensor_id', 'survey_id')
+
+
+class SensorReplacements(models.Model):
+    sensor_id = models.IntegerField()
+    old_hardware_id = models.IntegerField()
+    old_survey_device_id = models.IntegerField()
+    old_survey_id = models.IntegerField()
+    new_hardware_id = models.IntegerField()
+    new_survey_device_id = models.IntegerField()
+    new_survey_id = models.IntegerField()
+    datetime = models.DateTimeField()
+
+    def __str__(self):
+        return f"Sensor replacement object, {self.sensor_id}: {self.datetime}"
+
+    class Meta:
+        _DATABASE = 'default'
+
+# OccupEye ruined a great algorithm because they allow sensors to exist
+# in two different locations at exactly the same time, therefore we need
+# both sensor_id and survey_id
+
+
+class Historical(models.Model):
+    sensor_id = models.IntegerField()
+    survey_id = models.IntegerField()
+    datetime = models.DateTimeField()
+    state = models.IntegerField()
```

Comment:
When looking at historical values, the state will always be `-1` before the sensor exists and sometimes after the sensor stops existing (although if repurposed it will default to `0`). `-1` also occurs if the sensor was not accessible etc. so it can be quite useful to know explicitly if something has gone wrong (and we are only storing the deltas so a missing entry is not enough).